### PR TITLE
PR UX, diffs in sidebar

### PR DESCRIPTION
**Pull request by eric-j**
New PR UX, JB inline commenting

### CODE REVIEW

File: shared/ui/Stream/OpenPullRequests.tsx [hunk 1]
```diff
@@ -104,13 +130,23 @@ const Root = styled.div`
 			margin-top: 0px;
 		}
 	}
+	.files-changed-list-dropdown {
```
Comment (eric-j) [inline shared/ui/Stream/OpenPullRequests.tsx:133]: This is just at test comment... testing PRs on a PR for PRs... woah

File: shared/ui/Stream/OpenPullRequests.tsx [hunk 2]
```diff
@@ -104,13 +130,23 @@ const Root = styled.div`
 			margin-top: 0px;
 		}
```
Comment (eric-j) [inline shared/ui/Stream/OpenPullRequests.tsx:131]: testtestteststet

File: shared/ui/Stream/OpenPullRequests.tsx [hunk 3]
```diff
@@ -263,13 +322,17 @@ export const OpenPullRequests = React.memo((props: Props) => {
 	const [loadFromUrlQuery, setLoadFromUrlQuery] = React.useState({});
 	const [loadFromUrlOpen, setLoadFromUrlOpen] = React.useState("");
 	const [prError, setPrError] = React.useState("");
```
Comment (eric-j) [inline shared/ui/Stream/OpenPullRequests.tsx:324]: another test comment

File: shared/ui/Stream/OpenPullRequests.tsx [hunk 4]
```diff
@@ -639,6 +703,218 @@ export const OpenPullRequests = React.memo((props: Props) => {
 		return total;
 	}, [pullRequestGroups]);
 
+	const clickPR = pr => {
+		// if we have an expanded PR diffs in the sidebar, collapse it
+		if (pr.id === derivedState.expandedPullRequestId) {
+			dispatch(clearCurrentPullRequest());
+		} else {
+			// otherwise, either open the PR details or show the diffs,
+			// depending on the user's preference
```
Comment (eric-j) [inline shared/ui/Stream/OpenPullRequests.tsx:712]: and another

File: shared/ui/Stream/OpenPullRequests.tsx [hunk 5]
```diff
@@ -639,6 +703,218 @@ export const OpenPullRequests = React.memo((props: Props) => {
...
+
+			fetchOnePR(pr.providerId, pr.id);
+
+			HostApi.instance.track("PR Clicked", {
+				Host: pr.providerId
+			});
+		}
+	};
+
+	const fetchOnePR = async (providerId: string, pullRequestId: string, message?: string) => {
```
Comment (eric-j) [inline shared/ui/Stream/OpenPullRequests.tsx:724]: test

File: shared/ui/Stream/OpenPullRequests.tsx (hunk 2 above)
Comment (eric-j) [reply shared/ui/Stream/OpenPullRequests.tsx:132]: test 2

Comment (eric-j) [reply shared/ui/Stream/OpenPullRequests.tsx:132]: test 3

File: shared/ui/Stream/OpenPullRequests.tsx (hunk 1 above)
Comment (eric-j) [reply shared/ui/Stream/OpenPullRequests.tsx:134]: testtttt

Comment (eric-j) [reply shared/ui/Stream/OpenPullRequests.tsx:134]: test 2222

File: shared/ui/Stream/Review/ChangesetFile.tsx [hunk 6]
```diff
@@ -11,11 +11,13 @@ interface Props {
 	selected?: boolean;
 	noHover?: boolean;
 	icon?: any;
+	iconLast?: any;
 	actionIcons?: any;
 	tooltip?: any;
 	depth?: number;
 	viewMode?: "files" | "tree";
 	badge?: React.ReactNode;
+	chevron?: any;
```
Comment (eric-j) [inline shared/ui/Stream/Review/ChangesetFile.tsx:20]: test

**Review by bcanzanella — CHANGES_REQUESTED** [review]
couple tiny things, but this looks incredible, great work!

**Review by bcanzanella — APPROVED** [review]
this is killer! nice job!

**Comment by bcanzanella** [comment]
we'll leave all this in the branch for now... can send beta users vsix from this branch in TC.

File: shared/ui/store/context/reducer.ts [hunk 7]
```diff
@@ -158,7 +158,10 @@ export function reduceContext(
 								providerId: action.payload.providerId,
 								id: action.payload.id,
 								commentId: action.payload.commentId,
-								source: action.payload.source
+								source: action.payload.source,
+								view: action.payload.view,
+								previousView: state?.currentPullRequest?.view,
+								groupIndex: action.payload?.groupIndex
```
Comment (eric-j) [inline shared/ui/store/context/reducer.ts:164]: test

File: shared/ui/store/context/actions.ts [hunk 8]
```diff
@@ -185,8 +185,18 @@ export const setCurrentPullRequest = (
 	providerId: string,
 	id: string,
 	commentId?: string,
-	source?: string
-) => action(ContextActionsType.SetCurrentPullRequest, { providerId, id, commentId, source });
+	source?: string,
+	view?: "details" | "sidebar-diffs",
+	groupIndex?: string
+) =>
+	action(ContextActionsType.SetCurrentPullRequest, {
```
Comment (eric-j) [inline shared/ui/store/context/actions.ts:192]: hello world

File: jb/src/main/kotlin/editor/EditorComponentInlaysManager.kt [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+import com.intellij.openapi.editor.ex.util.EditorUtil
+import com.intellij.openapi.editor.impl.EditorEmbeddedComponentManager
+import com.intellij.openapi.editor.impl.EditorImpl
+import com.intellij.openapi.editor.impl.view.FontLayoutService
+import com.intellij.openapi.util.Disposer
+import com.intellij.ui.components.JBScrollPane
+import com.intellij.util.concurrency.annotations.RequiresEdt
+import com.intellij.util.ui.JBUI
+import com.intellij.util.ui.components.BorderLayoutPanel
+import java.awt.Dimension
```
Comment (eric-j) [inline jb/src/main/kotlin/editor/EditorComponentInlaysManager.kt:14]: asd

**Review by cstryker — COMMENTED** [review]
FYI, i'm sort of abstaining from this review since it's more or less out of my wheelhouse ... i can still review though, if needed

**Pull request merged**